doc/remediation-templating.md

```diff
@@ -0,0 +1,342 @@
+    ignition:
+      version: 3.1.0
+    storage:
+      files:
+      - contents:
+          source: data:, `PrintLastLog%20yes%0A%23TCPKeepAlive%20yes%0APermitUserEnvironment%20no%0ACompression
+          %20no%0AClientAliveInterval%203600%0AClientAliveCountMax%200%0A%0A%23UseDNS%2
+          0no%0A%23PidFile%20%2Fvar%2Frun%2Fsshd.pid%0A%23MaxStartups%2010%3A30%3A100%0A%23PermitTunnel%20no`
+        mode: 0600
+        path: /etc/ssh/sshd_config
+        overwrite: true
+```
+#### Content writer user-flow
+
+As a content writer:
+It is now possible to write remediation content for kubenetes and use the XCCDF values from
+the rules file.
+
+Contiune with this rule as an example:
+
+```yaml
+documentation_complete: true
+title: 'Set SSH Idle Timeout Interval'
+...
+requires:
+    - sshd_set_keepalive
+
+ocil_clause: 'it is commented out or not configured properly'
+
+ocil: |-
+    Run the following command to see what the timeout interval is:
+    <pre>$ sudo grep ClientAliveInterval /etc/ssh/sshd_config</pre>
+    If properly configured, the output should be:
+    <pre>ClientAliveInterval {{{ xccdf_value("sshd_idle_timeout_value") }}}</pre>
+...
+```
+
+First, the content writer can first write a rule, if they want to use XCCDF Value,
+they need to defind the value in the `ocil`, the formate will be like  
+`{{{ xccdf_value("the_value_you_want_to_define") }}}` and use a value variable
+in the rules, and then they can use the XCCDF value `the_value_you_want_to_define` in their 
+remediations contents. Ideally, there is some steps that a Marco script needs to do:
+
+1. take `the_value_you_want_to_define` in remediation content and then replace the corresponding values to `VAR_THE_VALUE_WANT_TO_DEFINE`
+2. url-encode the whole thing 
+3. substitue `VAR_THE_VALUE_WANT_TO_DEFINE` to `{{"the_value_you_want_to_define" | findValue | urlquery}}` and generate kube remediaiton
+
+
+After the content build, the machine config object will have the custom XCCDF value in this formate
+`data:,...some-url-encoding{{"the_value_you_want_to_define" | findValue | urlquery}}some-url-encoding` in kube remediation xml.
+
+The real values sustitution will happen in the compliance operator
+after the Tailor scan, go template function in the operator do the substitution from `data:,...some-url-encoding{{"the_value_you_want_to_define" | findValue | urlquery}}some-url-encoding` to `data:,...some-url-encoding urlencoded(SOMESETVALUES) some-url-encoding`
+
+The post-processed `ComplianceRemediation` kube object will look like following:
+```yaml
+apiVersion: compliance.openshift.io/v1alpha1
+kind: ComplianceRemediation
+metadata:
+  selfLink: >-
+    /apis/compliance.openshift.io/v1alpha1/namespaces/openshift-compliance/complianceremediations/nist-moderate-modified-master-sshd-set-idle-timeout
+  resourceVersion: '35343'
+  name: nist-moderate-modified-master-sshd-set-idle-timeout
+  uid: 085f51e4-19bb-4478-9398-df409858dce6
+  creationTimestamp: '2021-07-01T19:06:43Z'
```

Comment:
From a user perspective, it would be nice to indicate that a variable was used to render the `MachineConfig`. We could add an annotation or something of the sort to indicate that. What do you think?

Reply:
Yes, I was going back an forth between listing all variables or only those that are missing but because remediations are so hard to read we should probably annotate with all variables used. There won't be that many in most cases.